src/sinks/prometheus/exporter.rs

```diff
@@ -509,6 +510,7 @@ impl StreamSink<Event> for PrometheusExporter {
                 }
             }
 
+            drop(finalizers);
```

Comment:
It would be nice to defer this until the metrics are actually fetched, but this is probably as good as we can reasonably do here, particularly since the acker is updated here as well.

Reply:
Right, yeah, the fetch is never guaranteed to come, so we can't possibly wait to finalize.